### PR TITLE
Cds 54508

### PR DESCRIPTION
**Pull request by tmacari-harness**
## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- Sub Category Field Check: `trigger subcategoryfieldcheck`

### CODE REVIEW

**Comment by tmacari-harness** [comment]
trigger subcategoryfieldcheck

File: docs/resources/platform_connector_terraform_cloud.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+### Read-Only
+
+- `id` (String) The ID of this resource.
+
+<a id="nestedblock--credentials"></a>
+### Nested Schema for `credentials`
+
+Required:
+
+- `api_token` (Block List, Min: 1, Max: 1) API token credentials to use for authentication. (see [below for nested schema](#nestedblock--credentials--api_token))
```
Comment (deepakbolangady) [inline docs/resources/platform_connector_terraform_cloud.md:75]:
```suggestion
- `api_token` (Block List, Min: 1, Max: 1) API token credentials to use for authentication. (See [below for nested schema](#nestedblock--credentials--api_token).)
```

File: docs/resources/platform_connector_terraform_cloud.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+<a id="nestedblock--credentials--api_token"></a>
+### Nested Schema for `credentials.api_token`
+
+Required:
+
+- `api_token_ref` (String) Reference to a secret containing the api token to use for authentication. To reference a secret at the organization scope, prefix 'org' to the expression: org.{identifier}. To reference a secret at the account scope, prefix 'account` to the expression: account.{identifier}.
```
Comment (deepakbolangady) [inline docs/resources/platform_connector_terraform_cloud.md:82]:
```suggestion
- `api_token_ref` (String) Reference to a secret containing the API token to use for authentication. To reference a secret at the organization scope, prefix 'org' to the expression: org.{identifier}. To reference a secret at the account scope, prefix 'account` to the expression: account.{identifier}.
```

File: docs/resources/platform_connector_terraform_cloud.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+## Import
+
+Import is supported using the following syntax:
+
+```shell
+# Import using terrafrm cloud provider connector id
```
Comment (deepakbolangady) [inline docs/resources/platform_connector_terraform_cloud.md:89]:
```suggestion
# Import using the Terraform Cloud provider connector ID.
```

File: examples/resources/harness_platform_connector_terraform_cloud/resource.tf [hunk 4]
```diff
@@ -0,0 +1,30 @@
+# Create Terraform Cloud connector using API token as secret
```
Comment (deepakbolangady) [inline examples/resources/harness_platform_connector_terraform_cloud/resource.tf:1]: Same comment about "using" (either write "by using" or "that uses", depending on which one is accurate.

File: examples/resources/harness_platform_connector_terraform_cloud/import.sh [hunk 5]
```diff
@@ -0,0 +1,2 @@
+# Import using terrafrm cloud provider connector id
```
Comment (deepakbolangady) [inline examples/resources/harness_platform_connector_terraform_cloud/import.sh:1]:
```suggestion
# Import using the Terraform Cloud provider connector ID.
```

File: examples/resources/harness_platform_connector_terraform_cloud/resource.tf [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  tags               = ["foo:bar"]
+  delegate_selectors = ["harness-delegate"]
+  credentials {
+    api_token {
+      api_token_ref = "account.TEST_terraform_cloud_api_token"
+    }
+  }
+}
+
+# Add connectivity mode by providing execute_on_delegate value. Default is to execute on Delegate
```
Comment (deepakbolangady) [inline examples/resources/harness_platform_connector_terraform_cloud/resource.tf:16]: Same as earlier.

File: internal/service/platform/connector/terraform_cloud.go [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+	"github.com/harness/harness-go-sdk/harness/nextgen"
+	"github.com/harness/terraform-provider-harness/helpers"
+	"github.com/harness/terraform-provider-harness/internal/utils"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceConnectorTerraformCloud() *schema.Resource {
+	resource := &schema.Resource{
+		Description:   "Resource for creating a TerraformCloud connector.",
```
Comment (deepakbolangady) [inline internal/service/platform/connector/terraform_cloud.go:16]:
```suggestion
		Description:   "Resource for creating a Terraform Cloud connector.",
```

File: internal/service/platform/connector/terraform_cloud.go [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+		ReadContext:   resourceConnectorTerraformCloudRead,
+		CreateContext: resourceConnectorTerraformCloudCreateOrUpdate,
+		UpdateContext: resourceConnectorTerraformCloudCreateOrUpdate,
+		DeleteContext: resourceConnectorDelete,
+		Importer:      helpers.MultiLevelResourceImporter,
+		Schema: map[string]*schema.Schema{
+			"url": {
+				Description: "URL of Terraform Cloud platform.",
```
Comment (deepakbolangady) [inline internal/service/platform/connector/terraform_cloud.go:24]:
```suggestion
				Description: "URL of the Terraform Cloud platform.",
```

File: internal/service/platform/connector/terraform_cloud.go [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"credentials": {
+				Description: "Credentials to connect to Terraform Cloud platform.",
```
Comment (deepakbolangady) [inline internal/service/platform/connector/terraform_cloud.go:29]:
```suggestion
				Description: "Credentials to connect to the Terraform Cloud platform.",
```

Comment (deepakbolangady) [reply internal/service/platform/connector/terraform_cloud.go:29]: It looks like the text beyond this point is the same as the text I've already commented on. Please incorporate as necessary. I'll end my review here. Thank you.

**Review by deepakbolangady — DISMISSED** [review]
Approving with the understanding that you will incorporate the changes. Let me know if an approval is required again. Thank you.

**Comment by tmacari-harness** [comment]
Hi @deepakbolangady, the above docs were generated automatically by a tool.
@micahlmartin should we stick to generated docs or we can adjust them?
cc @rathodmeetsatish

File: docs/resources/platform_connector_terraform_cloud.md [hunk 10]
```diff
@@ -0,0 +1,91 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "harness_platform_connector_terraform_cloud Resource - terraform-provider-harness"
+subcategory: "Next Gen"
```
[the remark this answers is not in the document]
Comment (tmacari-harness) [reply docs/resources/platform_connector_terraform_cloud.md:4]: This cant be changed now

File: docs/data-sources/platform_connector_terraform_cloud.md [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+- `id` (String) The ID of this resource.
+- `tags` (Set of String) Tags to associate with the resource.
+- `url` (String) URL of the Terraform Cloud platform.
+
+<a id="nestedatt--credentials"></a>
+### Nested Schema for `credentials`
+
+Read-Only:
+
+- `api_token` (List of Object) (see [below for nested schema](#nestedobjatt--credentials--api_token))
```
[the remark this answers is not in the document]
Comment (tmacari-harness) [reply docs/data-sources/platform_connector_terraform_cloud.md:45]: Those are changes done by docs autogenerating tool

File: internal/service/platform/connector/terraform_cloud_test.go [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
```
Comment (rathodmeetsatish) [inline internal/service/platform/connector/terraform_cloud_test.go:49]: please add a test similar to TestAccResourceConnectorJenkins_DeleteUnderlyingResource.
It is to test scenario when someone creates a resource via terraform but deletes it from ui and again applys from terraform then it should not error out but should create a new resource

**Review by rathodmeetsatish — CHANGES_REQUESTED** [review]

File: internal/service/platform/connector/terraform_cloud_test.go (hunk 12 above)
Comment (tmacari-harness) [reply internal/service/platform/connector/terraform_cloud_test.go:82]: Done

**Review by rathodmeetsatish — APPROVED** [review]

**Pull request merged**